### DISCUSSION TITLE
HTML: Meta variables for the source file and URL path of the currently rendered *.qmd file?

### DISCUSSION THREAD

**Discussion by HenrikBengtsson** [quarto-dev/quarto-cli#5029] [Q&A]
Hello, is there a meta variables or similar that gets me the path of the rendered source file, or the rendered HTML URL?  Something like: `{{< meta source-file >}}`, or `{{< meta book.html.page-url-path >}}`, or similar.  

Basically, I'd like to be able being able to generate URLs myself similar to the 'View source' and 'Edit this page' URLs created when specifying `repo-site` for a Quarto Book, e.g. adding `Source file: {{< meta source-file >}}` to the footer.  I'd also to add a permanent URL link to the footer for the rendered URL, e.g. `Share link: {{< meta book.site-url >}}/{{< meta book.html.page-url-path >}}`.

BTW, is there a way to list all available `{{< meta ... >}}` variables?  Or is that a documented list of the most common ones?  All I found in the official docs was <https://quarto.org/docs/authoring/variables.html#meta>.

**Comment by cderv** [comment]
I think the discussion in https://github.com/quarto-dev/quarto-cli/issues/2249 could be interesting to this. Especially, we expose in Lua filter some information (but still need to document #4499) 

Though, you would like to access this directly in the doc through shortcodes and not in a Lua filter. This is currently possible using an extension https://github.com/mcanouil/quarto-lua-env (if still working) and FR to include such feature inside Quarto is in https://github.com/quarto-dev/quarto-cli/issues/4770 that you can follow 

Nothing regarding accessible the page url though. @dragonstyle can we make this accessible in Lua somehow ?

**Reply by cscheid** [reply]
## What I'm about to describe here is almost certainly going to be broken in the future, so keep it in mind before you use it and then need to fix your filters!

If you want to know everything that is available as an option for the meta shortcode, you can do this:

- download the git version of quarto
- edit the file src/resources/filters/quarto-pre/options.lua in line 5 to remove the keyword `local`. Now, `allOptions` is a global variable for our Lua filters
- in a Lua filter, print that value using `quarto.log.output(allOptions)` inside a `Pandoc` handler.

Those are the values that the `meta` shortcode responds to.

But, let me repeat myself:

## This voids your warranty! There is no guarantee that the values here will be available from one quarto point release to the next! Please don't depend on this behavior unless you're ready for it to break!

I do mean it! 😆 As I wrote in #4770, we consider many of these options to not be a public API (and unfortunately we are knee-deep in other things that stop us from documenting what we consider to be a public, stable API)
